clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -831,6 +831,7 @@ public void testSubscriptionChangesWithAutoCommitEnabled() {
         assertTrue(consumer.subscription().isEmpty());
         assertTrue(consumer.assignment().isEmpty());
 
+        client.requests().clear();
```

Comment:
@ijuma Thank you for the review. Yes, `clear` is required to avoid `close()` waiting for the timeout. Since these tests use `MockTime`, the tests would otherwise need to be updated to change time during `close()`. With this change, the timeout code path was no longer being triggered in unit tests, so I have added a couple of tests.